src/pages/List/List.js

```diff
@@ -1,5 +1,181 @@
+      '{"' +
+        decodeURI(queryString)
+          .replace(/"/g, '\\"')
+          .replace(/&/g, '","')
+          .replace(/=/g, '":"') +
+        '"}'
+    );
+  }
+
+  let queryStringData = { ...queryStringObject, ...filterData };
+  let filterQueryString = '';
+  for (const key in queryStringData) {
+    let queryString = '';
+    if (typeof queryStringData[key] === 'string') {
+      queryString += `&${key}=${queryStringData[key]}`;
+    } else {
+      for (let i = 0; i < queryStringData[key].length; i++) {
+        if (key === 'sort') {
+          queryString += `&${key}=${sortValue[queryStringData[key][i]]}`;
+        } else if (key === 'difficulty' && queryStringData[key].length < 3) {
+          queryString += `&${key}=${queryStringData[key][i]}`;
+        }
+      }
+    }
+    filterQueryString += queryString;
+  }
+  const queryStringResult = filterQueryString.substring(1);
+
+  useEffect(() => {
+    navigate(`?${queryStringResult}`);
+    fetch('/data/listData.json?' + queryStringResult)
+      .then(res => res.json())
+      .then(res => setClassData(res));
+  }, [queryStringResult, navigate]);
+
+  const openModal = e => {
+    let name;
+    if (e.target.innerText === '난이도') {
+      name = 'difficulty';
+    } else {
+      name = 'sort';
+    }
+    setModalOpen(prev => ({ ...prev, [name]: true }));
+  };
+
+  const closeModal = () => {
+    setModalOpen({
+      sort: false,
+      difficulty: false,
+    });
+  };
+
+  useOutsideClick(ref, () =>
+    setModalOpen({
+      sort: false,
+      difficulty: false,
+    })
+  );
+
+  const clickFilter = e => {
+    const { title, innerText } = e.target;
+    setFilterData(prev => ({ ...prev, [title]: [innerText] }));
+    closeModal();
+  };
+
+  return (
+    classData.length > 0 && (
+      <div>
+        {FILTER_DATA.map((data, index) => {
+          return (
+            <ModalWrap
+              key={index}
+              modalOpen={modalOpen}
+              closeModal={closeModal}
+              clickFilter={clickFilter}
+              filterData={filterData}
+              setFilterData={setFilterData}
+              ref={el => (ref.current[index] = el)}
+              title={data.title}
+              list={data.list}
+            />
+          );
+        })}
+        <ListPage>
+          <FilterWrap>
+            <Filter
+              name={filterData.sort.length > 0 ? filterData.sort : '추천순'}
+              openModal={openModal}
+            />
+            <Filter name="난이도" openModal={openModal} />
+          </FilterWrap>
+          <ClassListWrap>
+            {classData.map((data, index) => {
+              let result;
+              if (index < 8) {
+                result = (
+                  <Class
+                    key={data.id}
+                    data={data}
+                    classData={classData}
+                    setClassData={setClassData}
+                  />
+                );
+              }
+              return result;
```

Comment:
변수 선언 및 할당 없이도 할 수 있을 것 같습니다.
```suggestion
              index < 8 && return (
                  <Class
                    key={data.id}
                    data={data}
                    classData={classData}
                    setClassData={setClassData}
                  />
                );
```

Reply:
와.. 진짜 👏👏👏